### PR TITLE
NR-101030: enable automatic release

### PR DESCRIPTION
**Pull request by sigilioso**
~Blocked by: https://github.com/newrelic/coreint-automation/pull/13~

### CODE REVIEW

File: .github/.renovate.json5 [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "extends": [
+    "config:base"
+  ],
```
Comment (marcsanmi) [inline .github/.renovate.json5:4]: I would add the `"labels": ["dependencies"]` config as well.

File: .github/workflows/security.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  schedule:
+    - cron: "0 3 * * *"
+
+jobs:
+  trivy:
+    name: Trivy security scan
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (marcsanmi) [inline .github/workflows/security.yml:19]: I'd update it to`checkout@v3`

**Review by gsanchezgavier — COMMENTED** [review]
Looks good I just left a question.

File: .github/workflows/security.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: Run Trivy vulnerability scanner in repo mode
+        uses: aquasecurity/trivy-action@0.7.1
+        if: ${{ ! github.event.schedule }} # Do not run inline checks when running periodically
+        with:
+          scan-type: fs
+          ignore-unfixed: true
+          exit-code: 1
+          severity: 'HIGH,CRITICAL'
+          skip-files: 'tests/integration/consumer-producer/pom.xml'
```
[the remark this answers is not in the document]
Comment (sigilioso) [reply .github/workflows/security.yml:29]: Thanks for pointing it out, it was a copy-paste issue 📋

File: .github/workflows/trigger_prerelease.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+# For more details about how to release follow https://github.com/newrelic/coreint-automation/blob/main/docs/release_runbook.md
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 15 * * 1"
+
+jobs:
+  prerelease:
+    uses: newrelic/coreint-automation/.github/workflows/trigger_prerelease.yaml@v1
```
Comment (sigilioso) [inline .github/workflows/trigger_prerelease.yml:14]: We may want to update this to ignore the `tests` folder when the `exclude-dirs` parameter is exposed.

File: .github/workflows/security.yml (hunk 2 above)
Comment (sigilioso) [reply .github/workflows/security.yml]: Updated 9a9f97ed4d344dc8fe98f60bf8e947edf9794dc2, thanks!

File: .github/workflows/security.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sigilioso) [reply .github/workflows/security.yml]: Updated here: 9a9f97ed4d344dc8fe98f60bf8e947edf9794dc2

File: .github/workflows/trigger_prerelease.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    secrets:
```
Comment (paologallinaharbur) [inline .github/workflows/trigger_prerelease.yml:15]: Currently we are not consistently sending a warn message whenever a release is failing.

Checking https://github.com/newrelic/nri-postgresql/blob/master/.github/workflows/prerelease.yml we do not do the same in the different steps and we do not always include 
```
❌ `${{ env.REPO_FULL_NAME }}`: prerelease pipeline failed.
```
Therefore, if the pre-release fails when running tests the only person notified is who created the preselease (the bot) and who modified the pipeline.


Is it the right approach to rely on such a message? Should we find something else?

Comment (paologallinaharbur) [reply .github/workflows/trigger_prerelease.yml:15]: (I am adding this here to open a discussion, but it is kind of un-related to the PR, we can move it to a different place)

Comment (paologallinaharbur) [reply .github/workflows/trigger_prerelease.yml:15]: I was referring to an other issue, but thanks also that is needed! 😄

Comment (sigilioso) [reply .github/workflows/trigger_prerelease.yml:15]: Right! I mixed them up! 😅

**Review by marcsanmi — APPROVED** [review]
I like the new approach! 👍

File: .github/workflows/prerelease.yml [hunk 6]
```diff
@@ -198,4 +183,16 @@ jobs:
           integration: nri-${{ env.INTEGRATION }}
           packageLocation: repo
           stagingRepo: true
-          upgrade: false
\ No newline at end of file
+          upgrade: false
+
+  notify-failure:
```
Comment (paologallinaharbur) [inline .github/workflows/prerelease.yml:188]: thanks for finding this! I would update the release job as well, no?

File: .github/workflows/prerelease.yml [hunk 7]
```diff
@@ -147,14 +140,6 @@ jobs:
         shell: bash
         run: |
           build/windows/upload_msi.sh ${INTEGRATION} ${{ matrix.goarch }} ${TAG}
-      - name: Notify failure via Slack
```
Comment (gsanchezgavier) [inline .github/workflows/prerelease.yml:150]: why removing this? i think this is working https://newrelic.slack.com/archives/C01516RVD4H/p1684393634080039

Comment (gsanchezgavier) [reply .github/workflows/prerelease.yml:150]: answered on slack thanks

**Review by gsanchezgavier — APPROVED** [review]
Not sure if you want to keep the pr open to include the missing filters but LGTM thanks!

**Comment by sigilioso** [comment]
> Not sure if you want to keep the pr open to include the missing filters but LGTM thanks!

I'll keep it open a bit longer and include the missing filter once [newrelic/coreint-automation](https://github.com/newrelic/coreint-automation/pull/12) is merged.

File: .github/workflows/prerelease.yml (hunk 6 above)
Comment (sigilioso) [reply .github/workflows/prerelease.yml:188]: I included it in the release job here: f85201d0aa525f4d739e96508498ecce76f053d1, thanks for pointing it out!

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -57,4 +57,16 @@ jobs:
...
+    if: ${{ always() && failure() }}
+    needs: [publish-to-s3]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Notify failure via Slack
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.COREINT_SLACK_TOKEN }}
+          slack-channel: ${{ secrets.COREINT_SLACK_CHANNEL }}
+          slack-text: "❌ `${{ env.REPO_FULL_NAME }}`: [release pipeline failed](${{ github.server_url }}/${{ env.ORIGINAL_REPO_NAME }}/actions/runs/${{ github.run_id }})."
```
Comment (paologallinaharbur) [inline .github/workflows/release.yml:72]: notice that in this case you are using two different identifiers REPO_FULL_NAME and ORIGINAL_REPO_NAME and the first one is not defined

Comment (sigilioso) [reply .github/workflows/release.yml]: Updated: [6778e8d](https://github.com/newrelic/nri-postgresql/pull/106/commits/6778e8d6cf4a1bacd8703e7f7bf51ce063225fba), thanks!

File: .github/workflows/security.yml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        uses: aquasecurity/trivy-action@0.7.1
+        if: ${{ github.event.schedule }} # Generate sarif when running periodically
+        with:
+          scan-type: fs
+          ignore-unfixed: true
+          severity: 'HIGH,CRITICAL'
+          format: 'template'
+          template: '@/contrib/sarif.tpl'
+          output: 'trivy-results.sarif'
+          skip-files: 'tests/integration/consumer-producer/pom.xml'
```
Comment (paologallinaharbur) [inline .github/workflows/security.yml:41]: copy paste?

Comment (sigilioso) [reply .github/workflows/security.yml]: Thanks for noticing, fixed: [74ac15f](https://github.com/newrelic/nri-postgresql/pull/106/commits/74ac15f726d55e2fa18472da809495de966c1193)

**Pull request merged**